docs/examples/Transformer Examples.ipynb

```diff
@@ -533,16 +527,1066 @@
     "\n",
     "# one-time state estimation\n",
     "output_data4 = model4.calculate_state_estimation(\n",
-    "    symmetric=True, \n",
-    "    error_tolerance=1e-8, \n",
-    "    max_iterations=20, \n",
-    "    calculation_method=CalculationMethod.iterative_linear\n",
+    "    symmetric=True, error_tolerance=1e-8, max_iterations=20, calculation_method=CalculationMethod.iterative_linear\n",
     ")\n",
     "\n",
     "# result dataset\n",
     "print(\"------three-winding transformer result------\")\n",
     "print(pd.DataFrame(output_data4[\"three_winding_transformer\"]))"
    ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Power flow calculations with automatic tap changing\n",
+    "\n",
+    "### Example Network\n",
+    "\n",
+    "We use a simple network with 3 nodes, 1 source, 1 load, 1 line, 1 transformer and 1 transformer tap regulator. As shown below:\n",
+    "\n",
+    "```txt\n",
+    "                   (tap_side)  (control side)\n",
+    "source_1 --- node_2 --- transformer_3 --- node_4 --- line_5 --- node_6 --- load_7\n",
+    "                        |                    |\n",
+    "        transformer_tap_regulator_8 <--------/ (control voltage)\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "id": "2d786f12",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from power_grid_model.enum import _ExperimentalFeatures\n",
+    "\n",
+    "\n",
+    "# Power flow with automatic tap changing is still experimental at the time of writing.\n",
+    "# Create a derived PowerGridModel that has experimental features enabled.\n",
+    "class PowerGridModelWithExperimentalFeatures(PowerGridModel):\n",
+    "    def __init__(self, *args, **kwargs):\n",
+    "        super().__init__(*args, **kwargs)\n",
+    "\n",
+    "    def calculate_power_flow(self, **kwargs):\n",
+    "        return super()._calculate_power_flow(experimental_features=_ExperimentalFeatures.enabled, **kwargs)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "5ab5adba",
+   "metadata": {},
+   "source": [
+    "### Input Dataset\n",
+    "\n",
+    "We create an input dataset by using the helper function `initialize_array`. \n",
+    "\n",
+    "Please refer to [Components](../user_manual/components.md) for detailed explanation of all component types and their input/output attributes."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "id": "0c89a335",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# node\n",
+    "node = initialize_array(\"input\", \"node\", 3)\n",
```

Comment:
We probably can do a copy here as well from `input_data`. And in description above this cell, refer to `#input-dataset` section

Reply:
unfortunately not because we need 3 nodes here instead of 2. alternatively, we can copy the ones from `input_data` here but then we would need to create a full new one with 3 nodes for the line drop compensation. I think it's better to only do one input data set for the auto tap changing entirely. I would propose to keep it as is